toolkit/tools/scheduler/schedulerutils/preparerequest.go

```diff
@@ -149,13 +158,20 @@ func partnerTestNodesToRequest(pkgGraph *pkggraph.PkgGraph, buildState *GraphBui
 		return
 	}
 
-	return &BuildRequest{
+	request = &BuildRequest{
 		Node:           testNode,
 		PkgGraph:       pkgGraph,
 		AncillaryNodes: ancillaryTestNodes,
 		IsDelta:        isDelta,
-		UseCache:       buildUsesCache && canUseCacheForNode(pkgGraph, testNode, testsToRerun, buildState),
+		Freshness:      buildState.GetMaxFreshness(),
 	}
+
+	requiredRebuild := isRequiredRebuild(request.Node, testsToRerun)
+	if !requiredRebuild && buildUsesCache {
+		// We might be able to use the cache, set the freshness based on node's dependencies.
+		request.UseCache, request.Freshness = canUseCacheForNode(pkgGraph, testNode, buildState)
+	}
+	return
```

Comment:
Looks like I missed that while working on the test nodes change: could I ask you to replace this repeated code with a call to `buildRequest`? It seems like it's identical in terms of what it does.

Reply:
Will do